### PR TITLE
Add ProcFS

### LINKED ISSUE #270: Support ProcFS
The proc file system acts as an interface to internal data structures in the kernel, more info can be found [here](https://www.kernel.org/doc/Documentation/filesystems/proc.txt).

We can implement an FS called procfs which provides several frequently-used interfaces to Apps.

### PR DESCRIPTION
**Pull request by liqinggd**
Fixes #270

### CODE REVIEW

**Comment by tatetian** [comment]
There are quite a few duplicated code in the PR. Try to reduce the duplications :)

**Review by tatetian — COMMENTED** [review]
Looks good. Just needs to wait for the CI fix (see the line comments above).

File: .github/workflows/demo_test.yml
Comment (jessehui) [reply .github/workflows/demo_test.yml:379]: This line can be used to make sure the demo can pass. However, before this commit is actually merged, this line can be removed because if we changed the OpenVINO demo, a new OpenVINO demo image will/should be created and used in the successive tests.

**Review by tatetian — APPROVED** [review]
LGTM

**Pull request merged**
